### PR TITLE
veloren 0.13.0 (new formula)

### PR DESCRIPTION
**Pull request by printfn**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/veloren.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class Veloren < Formula
+  desc "Multiplayer voxel RPG"
+  homepage "https://veloren.net"
+  url "https://gitlab.com/veloren/veloren.git",
+    tag:      "v0.13.0",
+    revision: "34ae5b9df73f367fa34350337ebba71711f48b86"
```
Comment (carlocab) [inline Formula/veloren.rb:6]:
```suggestion
  url "https://gitlab.com/veloren/veloren.git",
      tag:      "v0.13.0",
      revision: "34ae5b9df73f367fa34350337ebba71711f48b86"
```

File: Formula/veloren.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  license "GPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "git-lfs" => :build
+  depends_on "ninja" => :build
+  depends_on "rustup-init" => :build
```
Comment (carlocab) [inline Formula/veloren.rb:12]: This needs to build with the Rust formula. While we do have some formulae that use `rustup` to build, that's just a workaround to get existing formulae working that we don't want to adopt for new formulae.

File: Formula/veloren.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    depends_on "pkg-config" => :build
+    depends_on "alsa-lib"
+    depends_on "libxcb"
+    depends_on "libxkbcommon"
+    depends_on "systemd" # for libudev
+  end
+
+  def install
+    system "git", "lfs", "install", "--local"
+    system "git", "lfs", "pull"
```
Comment (carlocab) [inline Formula/veloren.rb:25]: Why is this needed?

File: Formula/veloren.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    # make sure the assets have been downloaded correctly with git-lfs
+    unless `file assets/voxygen/background/bg_main.jpg`.match?(/JPEG image/i)
+      odie "assets have not been downloaded correctly"
```
Comment (carlocab) [inline Formula/veloren.rb:29]: We should use `resource` blocks to download these assets.

File: Formula/veloren.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    ENV["VELOREN_USERDATA_STRATEGY"] = "system"
+
+    system "cargo", "install", *std_cargo_args(root: libexec, path: "voxygen")
+    system "cargo", "install", *std_cargo_args(root: libexec, path: "server-cli")
+
+    libexec.install "assets"
+
+    (bin/"veloren-voxygen").write_env_script "#{libexec}/bin/veloren-voxygen",
+                                             VELOREN_ASSETS: "#{libexec}/assets"
+    (bin/"veloren-server-cli").write_env_script "#{libexec}/bin/veloren-server-cli",
```
Comment (carlocab) [inline Formula/veloren.rb:45]:
```suggestion
    (bin/"veloren-voxygen").write_env_script libexec/"bin/veloren-voxygen",
                                             VELOREN_ASSETS: libexec/"assets"
    (bin/"veloren-server-cli").write_env_script libexec/"bin/veloren-server-cli",
                                                VELOREN_ASSETS: libexec/"assets"
```

File: Formula/veloren.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                                                VELOREN_ASSETS: "#{libexec}/assets"
+  end
+
+  def caveats
+    <<~EOS
+      To start Veloren, run `veloren-voxygen`.
+
+      A multiplayer server can be started by running `veloren-server-cli`.
+    EOS
```
Comment (carlocab) [inline Formula/veloren.rb:54]:
```suggestion
```
We only accept new `caveat`s when they are specific to Homebrew.

File: Formula/veloren.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  test do
```
Comment (carlocab) [inline Formula/veloren.rb:57]:
```suggestion
    output = pipe_output(bin/"veloren-server-cli", "shutdown graceful 0")
```

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/veloren.rb (hunk 2 above)
Comment (printfn) [reply Formula/veloren.rb:12]: Unfortunately it requires nightly rust, and uses a `rust-toolchain` file to pin a specific nightly version. Is there a way to achieve this without `rustup`?

File: Formula/veloren.rb (hunk 1 above)
Comment (printfn) [reply Formula/veloren.rb]: fixed

File: Formula/veloren.rb (hunk 5 above)
Comment (printfn) [reply Formula/veloren.rb]: fixed

File: Formula/veloren.rb (hunk 6 above)
Comment (printfn) [reply Formula/veloren.rb]: fixed

File: Formula/veloren.rb (hunk 7 above)
Comment (printfn) [reply Formula/veloren.rb:49]: fixed

File: Formula/veloren.rb (hunk 3 above)
Comment (printfn) [reply Formula/veloren.rb]: fixed

File: Formula/veloren.rb (hunk 4 above)
Comment (printfn) [reply Formula/veloren.rb]: fixed, now uses a `resource` block

File: Formula/veloren.rb (hunk 2 above)
Comment (carlocab) [reply Formula/veloren.rb:11]: I don't think so.

This seems like a really nice piece of software, but I find it hard to overlook not being able to build with the latest stable version of Rust.

I suggest hosting it in an external tap for now. If it proves to be popular, then it'll show up in our analytics data and that would at least justify including it here.

Comment (printfn) [reply Formula/veloren.rb:11]: That makes sense, thanks for the feedback! I'll go ahead and close this now.

Comment (carlocab) [reply Formula/veloren.rb:11]: From a quick look at the formula, I think it should be fine. But I'll need to take a closer look at the repo if/when you decide to open a PR.
